data-api/src/server/data-api/operations/findOneAndReplace.ts

```diff
@@ -0,0 +1,94 @@
+import { RouteParams, RouteResponse } from 'modelence/server';
+import { getDatabase } from '../mongoClient';
+import { processFilter, ErrorResponse } from '../utils';
+
+interface FindOneAndReplaceRequest {
+  dataSource: string;
+  database: string;
+  collection: string;
+  filter: Record<string, any>;
+  replacement: Record<string, any>;
+  projection?: Record<string, any>;
+  sort?: Record<string, any>;
+  upsert?: boolean;
+  returnNewDocument?: boolean;
+}
+
+interface FindOneAndReplaceResponse {
+  document: Record<string, any> | null;
+}
+
+export async function findOneAndReplace(params: RouteParams): Promise<RouteResponse<FindOneAndReplaceResponse | ErrorResponse>> {
+  try {
+    const {
+      dataSource,
+      database,
+      collection,
+      filter,
+      replacement,
+      projection,
+      sort,
+      upsert = false,
+      returnNewDocument = true
+    } = params.body as FindOneAndReplaceRequest;
+
+    // Validate required fields
+    if (!dataSource || !database || !collection || !filter || !replacement) {
+      return {
+        status: 400,
+        data: {
+          error: "Missing required fields: dataSource, database, collection, filter, replacement",
+          error_code: "InvalidParameter"
+        }
+      };
+    }
+
+    // Validate replacement document doesn't contain update operators
+    const hasUpdateOperator = Object.keys(replacement).some(key => key.startsWith('$'));
+    if (hasUpdateOperator) {
+      return {
+        status: 400,
+        data: {
+          error: "replacement document cannot contain update operators (keys starting with $)",
+          error_code: "InvalidParameter"
+        }
+      };
+    }
```

Comment:
Doesn't the built-in findOneAndReplace handle this?

Reply:
@artahian I've checked those error messages are better than the native ones, I will keep them for now

Reply:
(In this specific case it will just work) if we remove $ validation